internal/service/sesv2/email_identity_data_source.go

```diff
@@ -120,5 +121,5 @@ func dataSourceEmailIdentityRead(ctx context.Context, d *schema.ResourceData, me
 	d.Set("identity_type", string(out.IdentityType))
 	d.Set("verified_for_sending_status", out.VerifiedForSendingStatus)
 
-	return nil
+	return diags
```

Comment:
For my edification, why is this commit necessary? Do folks add diagnostics in non-error cases? Is this just setting up for a future where we may add diagnostics all throughout this function?

Reply:
The latter - future-proofing for cases where we might add non-error diagnostics. The newer Plugin-Framework [uses this diagnostics pattern](https://developer.hashicorp.com/terraform/plugin/framework/diagnostics), so we try to make Plugin-SDK based resources look similar. 

There are a bunch of legacy Plugin-SDK based resources that don't look this way, but when new stuff comes in its a simple enough change we usually switch over.